texinfo/info/info.man
.\" Copyright (c) 1991, 1992 Free Software Foundation
.\" See section COPYING for conditions for redistribution
.\" $Id: info.man,v 1.1 1993/02/26 02:00:08 zoo Exp $
.TH info 1 "3mar1992" "GNU Tools" "GNU Tools"
.de BP
.sp
.ti \-.2i
\(**
..

.SH NAME
info \- the GNU information browser

.SH SYNOPSIS
.hy 0
.na
.TP
.B info
.RB "[\|" "\-d "\c
.I path\c
.RB " | " "+directory "\c
.I path\c
\&\|]
.br
.RB "[\|" "\-n "\c
.I nodename\c
.RB ".\|.\|. | " "+node "\c
.I nodename\c
\&.\|.\|.\|]
.br
.RB "[\|" "\-f "\c
.I name\c
.RB " | " "+file "\c
.I name\c
\&\|]
.br
.RB "[\|" "\-o "\c
.I outfile\c
.RB " | " "+output "\c
.I outfile\c
\&\|]
.br
.RB "[\|" \c
.I topic\c
\&\|]

.ad b
.hy 1
.SH DESCRIPTION
A program to browse through GNU manuals online.  Its first display is a directory showing all the manuals available.  If
you already know which of those manuals you want, you can give the
corresponding menu entry as an argument to \c
.B info\c
\&; for example,
`\|\c
.B info info\c
\&\|' requests the online manual for \c
.B info\c
\& itself.

The manuals are organized into \c
.I nodes\c
\&, which correspond to the
chapters and sections of a printed book.  You can follow them in
sequence, just like in the printed book; but there are
also menus, ``hot'' references, and search facilities.

The best way to get started with \c
.B info
is to run it with no arguments.  The first screen displayed is
a menu of the documentation available.  Typing the single letter \c
.B h\c
\& requests a
tutorial.

GNU Emacs has a built-in facility equivalent to \c
.B info\c
\&.  If you
already use Emacs, you may want to get into the documentation browsing
mode, instead, by typing \c
.B C-h i\c
\& inside Emacs.

You can get out of \c
.B info\c
\& at any time by typing the single letter
\c
.B q\c
\&.  

Here is a summary of all the \c
.B info\c
\& commands; \c
.B info\c
\& itself
can display a summary like this at any time, when you type the single
character \c
.B ?\c
\&.

.ta \w'SpaceMMM'u
.B h\c
\&	Invoke the Info tutorial.
.br
.B q\c
\&	Quit Info

.I Selecting other nodes:
.br
.B n\c
\&	Move to the ``next'' node of this node.
.br
.B p\c
\&	Move to the ``previous'' node of this node.
.br
.B u\c
\&	Move ``up'' from this node.
.br
.B m\c
\&	Pick menu item specified by name (or abbreviation).
.br
.B f\c
\&	Follow a cross reference.  Reads name of reference.
.br
.B l\c
\&	Move to the last node you were looking at.
.br
.B d\c
\&	Display directory node.

.I Moving within a node:
.br
Space\c
\&	scroll forward a page.\c
.br
DEL\c
\&	scroll backward a page.\c
.br
.B b\c
\&	Go to beginning of this node.\c

.I Advanced commands:
.br
.B 1\c
\&	Pick first item in node's menu.\c
.br
.B 2 \- 5\c
\&	Pick second ... fifth item in node's menu.\c
.br
.B g\c
\&	Move to \c
.I nodename\c
\& or \c
.B (\c
.I filename\c
.B \&)\c
.I nodename\c
\&.
.br
.B s\c
\&	Search for specified string.
.br
Ctl-p\c
\&	Print the contents of this node.

.SH OPTIONS

.BI "-d " "path"
.TP
.BI "+directory " "path"\c
\&
A colon-separated list of directories; where to look for info files
and their directory.  The first of these directories containing a file
`\|\c
.B dir\c
\&\|' governs the top-level directory displayed; but you can
explicitly visit info files by name, whether or not they're in that
directory, from any directory in \c
.I path\c
\&.
.PP        
.BI "-n " "nodename"\c
.TP
.BI "+node " "nodename"\c
Go directly to node \c
.I nodename\c
\& of the current file.  Since
directories don't have nodes, this option is most useful if you also
specify a particular \c
.I topic\c
\& or use the `\|\c
.B +file\c
\&\|' option.  You
may specify more than one `\|\c
.B +node \c
.I nodename\c
\&\c
\&\|'; the rightmost
will be displayed, but you can visit the others in reverse sequence
using the command \c
.B l\c
\&.
.PP        
.BI "-f " "name"\c
.TP
.BI "+file " "name"\c
\&
Go directly to info file \c
.I name\c
\&, bypassing the main directory
display.  \c
.I name\c
\& may be a fully specified path; if not, it will be
sought through the info path directories.
.PP        
.BI "-o " "outfile"\c
.TP
.BI "+output " "outfile"\c
\&
Copy a node to file \c
.I outfile\c
\& instead of running interactively.
        
.TP
.I "topic"\c
\&
Go directly to topic \c
.I topic\c
\& from the main info directory.
.PP

.SH ENVIRONMENT

.TP
.B INFOPATH
Overrides the default paths compiled into `\|\c
.B info\c
\&\|' to specify where
to look for info files.  The `\|\c
.B +directory\c
\&\|' option in turn
overrides this environment variable.

.TP
.B INFO_PRINT_COMMAND
Overrides the compiled-in default printing command (normally
`\|\c
.B lpr\c
\&\|') used to print nodes on request.
.PP

.SH FILES
.ta \w'\fIINFODIR\fR\fB/dir  'u
\fIINFODIR\fR\fB/dir	Top-level directory menu
.br
\fIname\fR
.br
\fIname\fR\fB-info
.br
\fIname\fR\fB.info	tried (in sequence) when file \fIname\fR 
requested. 

.SH "SEE ALSO"
`\|\c
.B Info\c
\&\|', `\|\c
.B Emacs\c
\&\|', and `\|\c
.B Texinfo\c
\&\|' entries in \c
.B info\c
\&.
.br
.I GNU Emacs\c
, Richard M. Stallman, February 1988.
.br
.I 
Texinfo: the GNU Documentation Format\c
, Robert J. Chassell and Richard M. Stallman, February 1991.
.br
.BR emacs ( 1 ),
.BR makeinfo ( 1 ).
.SH ACKNOWLEDGEMENT
Richard Stallman invented the Info concept many moons ago, even before
the advent of hypertext.  He also suggested that Brian Fox write this program.
.SH AUTHOR
Brian J. Fox (bfox@ai.mit.edu)
.SH COPYING
Copyright (c) 1991, 1992 Free Software Foundation, Inc.
.PP
Permission is granted to make and distribute verbatim copies of
this manual provided the copyright notice and this permission notice
are preserved on all copies.
.PP
Permission is granted to copy and distribute modified versions of this
manual under the conditions for verbatim copying, provided that the
entire resulting derived work is distributed under the terms of a
permission notice identical to this one.
.PP
Permission is granted to copy and distribute translations of this
manual into another language, under the above conditions for modified
versions, except that this permission notice may be included in
translations approved by the Free Software Foundation instead of in
the original English.


gdb/NEWS
		What has changed since GDB-3.5?
		(Organized release by release)

*** Changes in GDB-4.10:

 * User visible changes:

Remote debugging using the GDB-specific (`target remote') protocol now
supports the `load' command.  This is only useful if you have some
other way of getting the stub to the target system, and you can put it
somewhere in memory where it won't get clobbered by the download.

Filename completion now works.

When run under emacs mode, the "info line" command now causes the
arrow to point to the line specified.  Also, "info line" prints
addresses in symbolic form (as well as hex).

All vxworks based targets now support a user settable option, called
vxworks-timeout.  This option represents the number of seconds gdb
should wait for responses to rpc's.  You might want to use this if
your vxworks target is, perhaps, a slow software simulator or happens
to be on the far side of a thin network line.

 * DEC alpha support

This release contains support for using a DEC alpha as a GDB host for
cross debugging.  Native alpha debugging is not supported yet.


*** Changes in GDB-4.9:

 * Testsuite

This is the first GDB release which is accompanied by a matching testsuite.
The testsuite requires installation of dejagnu, which should be available
via ftp from most sites that carry GNU software.

 * C++ demangling

'Cfront' style demangling has had its name changed to 'ARM' style, to
emphasize that it was written from the specifications in the C++ Annotated
Reference Manual, not necessarily to be compatible with AT&T cfront.  Despite
disclaimers, it still generated too much confusion with users attempting to
use gdb with AT&T cfront.

 * Simulators

GDB now uses a standard remote interface to a simulator library.
So far, the library contains simulators for the Zilog Z8001/2, the
Hitachi H8/300, H8/500 and Super-H.

 * New targets supported

H8/300 simulator 			h8300-hitachi-hms or h8300hms
H8/500 simulator 			h8500-hitachi-hms or h8500hms
SH simulator				sh-hitachi-hms    or sh
Z8000 simulator		  		z8k-zilog-none 	  or z8ksim
IDT MIPS board over serial line		mips-idt-ecoff

Cross-debugging to GO32 targets is supported.  It requires a custom
version of the i386-stub.c module which is integrated with the 
GO32 memory extender.

 * New remote protocols

MIPS remote debugging protocol.

 * New source languages supported

This version includes preliminary support for Chill, a Pascal like language
used by telecommunications companies.  Chill support is also being integrated
into the GNU compiler, but we don't know when it will be publically available.


*** Changes in GDB-4.8:

 * HP Precision Architecture supported

GDB now supports HP PA-RISC machines running HPUX.  A preliminary
version of this support was available as a set of patches from the
University of Utah.  GDB does not support debugging of programs
compiled with the HP compiler, because HP will not document their file
format.  Instead, you must use GCC (version 2.3.2 or later) and PA-GAS
(as available from jaguar.cs.utah.edu:/dist/pa-gas.u4.tar.Z).

Many problems in the preliminary version have been fixed.

 * Faster and better demangling

We have improved template demangling and fixed numerous bugs in the GNU style
demangler.  It can now handle type modifiers such as `static' or `const'.  Wide
character types (wchar_t) are now supported.  Demangling of each symbol is now
only done once, and is cached when the symbol table for a file is read in.
This results in a small increase in memory usage for C programs, a moderate
increase in memory usage for C++ programs, and a fantastic speedup in
symbol lookups.

`Cfront' style demangling still doesn't work with AT&T cfront.  It was written
from the specifications in the Annotated Reference Manual, which AT&T's
compiler does not actually implement.

 * G++ multiple inheritance compiler problem

In the 2.3.2 release of gcc/g++, how the compiler resolves multiple
inheritance lattices was reworked to properly discover ambiguities.  We
recently found an example which causes this new algorithm to fail in a
very subtle way, producing bad debug information for those classes.
The file 'gcc.patch' (in this directory) can be applied to gcc to
circumvent the problem.  A future GCC release will contain a complete
fix.

The previous G++ debug info problem (mentioned below for the gdb-4.7
release) is fixed in gcc version 2.3.2.

 * Improved configure script

The `configure' script will now attempt to guess your system type if
you don't supply a host system type.  The old scheme of supplying a
host system triplet is preferable over using this.  All the magic is
done in the new `config.guess' script.  Examine it for details.

We have also brought our configure script much more in line with the FSF's
version.  It now supports the --with-xxx options.  In particular,
`--with-minimal-bfd' can be used to make the GDB binary image smaller.
The resulting GDB will not be able to read arbitrary object file formats --
only the format ``expected'' to be used on the configured target system.
We hope to make this the default in a future release.

 * Documentation improvements

There's new internal documentation on how to modify GDB, and how to
produce clean changes to the code.  We implore people to read it
before submitting changes.

The GDB manual uses new, sexy Texinfo conditionals, rather than arcane
M4 macros.  The new texinfo.tex is provided in this release.  Pre-built
`info' files are also provided.  To build `info' files from scratch,
you will need the latest `makeinfo' release, which will be available in
a future texinfo-X.Y release.

*NOTE*  The new texinfo.tex can cause old versions of TeX to hang.
We're not sure exactly which versions have this problem, but it has
been seen in 3.0.  We highly recommend upgrading to TeX version 3.141
or better.  If that isn't possible, there is a patch in
`texinfo/tex3patch' that will modify `texinfo/texinfo.tex' to work
around this problem.

 * New features


texinfo/info/info.c
/* info.c -- Display nodes of Info files in multiple windows. */

#include "info.h"
#include "dribble.h"
#include "getopt.h"

/* The version numbers of this version of Info. */
int info_major_version = 2;
int info_minor_version = 9;
int info_patch_level = 1;

/* Non-zero means print version info only. */
static int print_version_p = 0;

/* Non-zero means print a short description of the options. */
static int print_help_p = 0;

/* Array of the names of nodes that the user specified with "--node" on the
   command line. */
static char **user_nodenames = (char **)NULL;
static int user_nodenames_index = 0;
static int user_nodenames_slots = 0;

/* String specifying the first file to load.  This string can only be set
   by the user specifying "--file" on the command line. */
static char *user_filename = (char *)NULL;

/* String specifying the name of the file to dump nodes to.  This value is
   filled if the user speficies "--output" on the command line. */
static char *user_output_filename = (char *)NULL;

/* Non-zero indicates that when "--output" is specified, all of the menu
   items of the specified nodes (and their subnodes as well) should be
   dumped in the order encountered.  This basically can print a book. */
int dump_subnodes = 0;

/* Structure describing the options that Info accepts.  We pass this structure
   to getopt_long ().  If you add or otherwise change this structure, you must
   also change the string which follows it. */
#define DRIBBLE_OPTION 2
#define RESTORE_OPTION 3
static struct option long_options[] = {
  { "directory", 1, 0, 'd' },
  { "node", 1, 0, 'n' },
  { "file", 1, 0, 'f' },
  { "subnodes", 0, &dump_subnodes, 1 },
  { "output", 1, 0, 'o' },
  { "help", 0, &print_help_p, 1 },
  { "version", 0, &print_version_p, 1 },
  { "dribble", 1, 0, DRIBBLE_OPTION },
  { "restore", 1, 0, RESTORE_OPTION },
  {NULL, 0, NULL, 0}
};

/* String describing the shorthand versions of the long options found above. */
static char *short_options = "d:n:f:o:s";

/* When non-zero, the Info window system has been initialized. */
int info_windows_initialized_p = 0;

/* Some "forward" declarations. */
static void usage (), info_short_help (), remember_info_program_name ();


/* **************************************************************** */
/*								    */
/*		  Main Entry Point to the Info Program		    */
/*								    */
/* **************************************************************** */

int
main (argc, argv)
     int argc;
     char **argv;
{
  int getopt_long_index;	/* Index returned by getopt_long (). */
  NODE *initial_node;		/* First node loaded by Info. */

#if defined (NeXT) && defined (NOTDEF)
  malloc_debug (0x0ffffffff);
#endif /* NeXT && NOTDEF */

  remember_info_program_name (argv[0]);

  while (1)
    {
      int option_character;

      option_character = getopt_long
	(argc, argv, short_options, long_options, &getopt_long_index);

      /* getopt_long () returns EOF when there are no more long options. */
      if (option_character == EOF)
	break;

      /* If this is a long option, then get the short version of it. */
      if (option_character == 0 && long_options[getopt_long_index].flag == 0)
	option_character = long_options[getopt_long_index].val;

      /* Case on the option that we have received. */
      switch (option_character)
	{
	case 0:
	  break;

	  /* User wants to add a directory. */
	case 'd':
	  info_add_path (optarg, INFOPATH_PREPEND);
	  break;

	  /* User is specifying a particular node. */
	case 'n':
	  add_pointer_to_array (optarg, user_nodenames_index, user_nodenames,
				user_nodenames_slots, 10, char *);
	  break;

	  /* User is specifying a particular Info file. */
	case 'f':
	  if (user_filename)
	    free (user_filename);

	  user_filename = savestring (optarg);
	  break;

	  /* User is specifying the name of a file to output to. */
	case 'o':
	  if (user_output_filename)
	    free (user_output_filename);
	  user_output_filename = savestring (optarg);
	  break;

	  /* User is specifying that she wishes to dump the subnodes of
	     the node that she is dumping. */
	case 's':
	  dump_subnodes = 1;
	  break;

	  /* User has specified a dribble file to receive keystrokes. */
	case DRIBBLE_OPTION:
	  close_dribble_file ();
	  open_dribble_file (optarg);
	  break;

	  /* User has specified an alternate input stream. */
	case RESTORE_OPTION:
	  info_set_input_from_file (optarg);
	  break;

	default:
	  usage ();
	}
    }

  /* If the user specified --version, then show the version and exit. */
  if (print_version_p)
    {
      printf ("GNU Info, Version %s.\n", version_string ());
      exit (0);
    }

  /* If the `--help' option was present, show the help and exit. */
  if (print_help_p)
    {
      info_short_help ();
      exit (0);
    }
  
  /* If the user hasn't specified a path for Info files, default that path
     now. */
  if (!infopath)
    {
      char *path_from_env, *getenv ();

      path_from_env = getenv ("INFOPATH");

      if (path_from_env)
	info_add_path (path_from_env);
      else
	info_add_path (DEFAULT_INFOPATH);
    }

  /* If the user specified a particular filename, add the path of that
     file to the contents of INFOPATH. */
  if (user_filename)
    {
      char *directory_name, *temp;

      directory_name = savestring (user_filename);
      temp = filename_non_directory (directory_name);

      if (temp != directory_name)
	{
	  *temp = 0;
	  info_add_path (directory_name, INFOPATH_PREPEND);
	}

      free (directory_name);
    }

  /* Get the initial Info node.  It is either "(dir)Top", or what the user
     specifed with values in user_filename and user_nodenames. */
  if (user_nodenames)
    initial_node = info_get_node (user_filename, user_nodenames[0]);
  else
    initial_node = info_get_node (user_filename, (char *)NULL);

  /* If we couldn't get the initial node, this user is in trouble. */
  if (!initial_node)
    {
      if (info_recent_file_error)
	info_error (info_recent_file_error);
      else
	info_error
	  (CANT_FIND_NODE, user_nodenames ? user_nodenames[0] : "Top");
      exit (1);
    }

  /* Special cases for when the user specifies multiple nodes.  If we are
     dumping to an output file, dump all of the nodes specified.  Otherwise,
     attempt to create enough windows to handle the nodes that this user wants
     displayed. */
  if (user_nodenames_index > 1)
    {
      free (initial_node);

      if (user_output_filename)
	dump_nodes_to_file
	  (user_filename, user_nodenames, user_output_filename, dump_subnodes);
      else
	begin_multiple_window_info_session (user_filename, user_nodenames);

      exit (0);
    }

  /* If there are arguments remaining, they are the names of menu items
     in sequential info files starting from the first one loaded.  That
     file name is either "dir", or the contents of user_filename if one
     was specified. */
  while (optind != argc)
    {
      REFERENCE **menu;
      REFERENCE *entry;
      NODE *node;
      char *arg;

      /* Remember the name of the menu entry we want. */
      arg = argv[optind++];

      /* Build and return a list of the menu items in this node. */
      menu = info_menu_of_node (initial_node);

      /* If there wasn't a menu item in this node, stop here, but let
	 the user continue to use Info.  Perhaps they wanted this node
	 and didn't realize it. */
      if (!menu)
	{
	  begin_info_session_with_error
	    (initial_node, "There is no menu in this node.");
	  exit (0);
	}

      /* Find the specified menu item. */
      entry = info_get_labeled_reference (arg, menu);

      /* If the item wasn't found, search the list sloppily.  Perhaps this
	 user typed "buffer" when they really meant "Buffers". */
      if (!entry)
	{
	  register int i;

	  for (i = 0; entry = menu[i]; i++)
	    if (strnicmp (entry->label, arg, strlen (arg)) == 0)
	      break;
	}

      /* If we failed to find the reference, start Info with the current
	 node anyway.  It is probably a misspelling. */
      if (!entry)
	{
	  char *error_message = "There is no menu item \"%s\" in this node.";

	  info_free_references (menu);

	  /* If we were supposed to dump this node, complain. */
	  if (user_output_filename)
	    info_error (error_message, arg);
	  else
	    begin_info_session_with_error (initial_node, error_message, arg);

	  exit (0);
	}

      /* We have found the reference that the user specified.  Clean it
	 up a little bit. */
      if (!entry->filename)
	{
	  if (initial_node->parent)
	    entry->filename = savestring (initial_node->parent);
	  else
	    entry->filename = savestring (initial_node->filename);
	}

      /* Find this node.  If we can find it, then turn the initial_node
	 into this one.  If we cannot find it, try using the label of the
	 entry as a file (i.e., "(LABEL)Top").  Otherwise the Info file is
	 malformed in some way, and we will just use the current value of
	 initial node. */
      node = info_get_node (entry->filename, entry->nodename);

      if (!node && entry->nodename &&
	  (strcmp (entry->label, entry->nodename) == 0))
	node = info_get_node (entry->label, "Top");

      if (node)
	{
	  free (initial_node);
	  initial_node = node;
	  info_free_references (menu);
	}
      else
	{
	  char *temp = savestring (entry->label);
	  char *error_message;

	  error_message = "Unable to find the node referenced by \"%s\".";

	  info_free_references (menu);

	  /* If we were trying to dump the node, then give up.  Otherwise,
	     start the session with an error message. */
	  if (user_output_filename)
	    info_error (error_message, temp);
	  else
	    begin_info_session_with_error (initial_node, error_message, temp);

	  exit (0);
	}
    }

  /* If the user specified that this node should be output, then do that
     now.  Otherwise, start the Info session with this node. */
  if (user_output_filename)
    dump_node_to_file (initial_node, user_output_filename, dump_subnodes);
  else
    begin_info_session (initial_node);

  exit (0);
}

/* Return a string describing the current version of Info. */
char *
version_string ()
{
  static char *vstring = (char *)NULL;

  if (!vstring)
    {
      vstring = (char *)xmalloc (50);
      sprintf (vstring, "%d.%d", info_major_version, info_minor_version);
      if (info_patch_level)
	sprintf (vstring + strlen (vstring), "-p%d", info_patch_level);
    }
  return (vstring);
}

/* **************************************************************** */
/*								    */
/*		   Error Handling for Info			    */
/*								    */
/* **************************************************************** */

static char *program_name = (char *)NULL;

static void
remember_info_program_name (fullpath)
     char *fullpath;
{
  char *filename;

  filename = filename_non_directory (fullpath);
  program_name = savestring (filename);
}

/* Non-zero if an error has been signalled. */
int info_error_was_printed = 0;

/* Non-zero means ring terminal bell on errors. */
int info_error_rings_bell_p = 1;

/* Print FORMAT with ARG1 and ARG2.  If the window system was initialized,
   then the message is printed in the echo area.  Otherwise, a message is
   output to stderr. */
void
info_error (format, arg1, arg2)
     char *format;
     void *arg1, *arg2;
{
  info_error_was_printed = 1;

  if (!info_windows_initialized_p || display_inhibited)
    {
      fprintf (stderr, "%s: ", program_name);
      fprintf (stderr, format, arg1, arg2);
      fprintf (stderr, "\n");
      fflush (stderr);
    }
  else
    {
      if (!echo_area_is_active)
	{
	  if (info_error_rings_bell_p)
	    terminal_ring_bell ();
	  window_message_in_echo_area (format, arg1, arg2);
	}
      else
	{
	  NODE *temp;

	  temp = build_message_node (format, arg1, arg2);
	  if (info_error_rings_bell_p)
	    terminal_ring_bell ();
	  inform_in_echo_area (temp->contents);
	  free (temp->contents);
	  free (temp);
	}
    }
}

/* Produce a very brief descripton of the available options and exit with
   an error. */
static void
usage ()
{
  fprintf (stderr,"%s\n%s\n%s\n%s\n%s\n",
"Usage: info [-d dir-path] [-f info-file] [-o output-file] [-n node-name]...",
"            [--directory dir-path] [--file info-file] [--node node-name]...",
"            [--help] [--output output-file] [--subnodes] [--version]",
"            [--dribble dribble-file] [--restore from-file]",
"            [menu-selection ...]");
  exit (1);
}

/* Produce a scaled down description of the available options to Info. */
static void
info_short_help ()
{
  printf ("%s", "\
Here is a quick description of Info's options.  For a more complete\n\
description of how to use Info, type `info info options'.\n\
\n\
   --directory DIR		Add DIR to INFOPATH.\n\
   --file FILENAME		Specify Info file to visit.\n\
   --node NODENAME		Specify nodes in first visited Info file.\n\
   --output FILENAME		Output selected nodes to FILENAME.\n\
   --dribble FILENAME		Remember user keystrokes in FILENAME.\n\
   --restore FILENAME		Read initial keystrokes from FILENAME.\n\
   --subnodes			Recursively output menu items.\n\
   --help			Get this help message.\n\
   --version			Display Info's version information.\n\
\n\
Remaining arguments to Info are treated as the names of menu\n\
items in the initial node visited.  You can easily move to the\n\
node of your choice by specifying the menu names which describe\n\
the path to that node.  For example, `info emacs buffers'.\n");

  exit (0);
}
